app/liquid_tags/codesandbox_tag.rb

```diff
@@ -14,18 +15,39 @@ def render(_context)
   private
 
   def parse_id(input)
-    input_no_space = input.delete(" ")
-    if valid_id?(input_no_space)
-      input_no_space
-    else
-      raise StandardError, "Invalid codesandbox Id"
+    id = input.split(" ").first
+    raise StandardError, "Invalid codesandbox ID" unless valid_id?(id)
 
-    end
+    id
   end
 
   def valid_id?(id)
     id =~ /\A[a-zA-Z0-9\-]{0,60}\Z/
   end
+
+  def parse_options(input)
+    _, *options = input.split(" ")
+
+    # Validation
+    validated_options = options.map { |o| valid_option(o) }.reject { |e| e == nil }
+    raise StandardError, "Invalid Options" unless options.empty? || !validated_options.empty?
```

Comment:
Let's show a more descriptive error:
```suggestion
    raise StandardError, "CodeSandbox Error: Invalid options" unless options.empty? || !validated_options.empty?
```

Reply:
@Zhao-Andy I agree with this suggestion, but looking at this further, I am thinking I should remove the `|| !validated_options.empty?` from the end of the line.  As coded, if there is one valid option and one bad option, the bad option is ignored and an exception is not raised.  Removing it will cause any bad option to raise an exception.  I think this might be better than silently ignoring it for some cases.

Thoughts?

Reply:
Decided to change the logic to raise an exception if any option is invalid.